File: lib/core/utils/constants/error_code.dart
import 'package:flutter_utils/core/utils/annotations/int_def.dart';

@IntDef([
  ErrorCode.CANCELED,
  ErrorCode.FAILURE,
  ErrorCode.NETWORK_UNAVAILABLE,
  ErrorCode.NULLABLE_OBJECT,
  ErrorCode.PAUSED,
  ErrorCode.RESULT_NOT_FOUND,
  ErrorCode.STOPPED,
  ErrorCode.TIME_OUT
])
class ErrorCode {
  static const int NONE = 10000;
  static const int CANCELED = 10010;
  static const int FAILURE = 10020;
  static const int NETWORK_UNAVAILABLE = 10030;
  static const int NULLABLE_OBJECT = 10040;
  static const int PAUSED = 10050;
  static const int RESULT_NOT_FOUND = 10060;
  static const int STOPPED = 10070;
  static const int TIME_OUT = 10080;
}

File: lib/core/utils/constants/ErrorCode.java
package com.picon.utils.constains;

import static com.picon.utils.constains.ErrorCode.CANCELED;
import static com.picon.utils.constains.ErrorCode.FAILURE;
import static com.picon.utils.constains.ErrorCode.NETWORK_UNAVAILABLE;
import static com.picon.utils.constains.ErrorCode.NULLABLE_OBJECT;
import static com.picon.utils.constains.ErrorCode.PAUSED;
import static com.picon.utils.constains.ErrorCode.RESULT_NOT_FOUND;
import static com.picon.utils.constains.ErrorCode.STOPPED;
import static com.picon.utils.constains.ErrorCode.TIME_OUT;

import androidx.annotation.IntDef;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.SOURCE)
@Documented
@IntDef({
        CANCELED,
        FAILURE,
        NETWORK_UNAVAILABLE,
        NULLABLE_OBJECT,
        PAUSED,
        RESULT_NOT_FOUND,
        STOPPED,
        TIME_OUT
})
public @interface ErrorCode {
    int CANCELED = 10010;
    int FAILURE = 10020;
    int NETWORK_UNAVAILABLE = 10030;
    int NULLABLE_OBJECT = 10040;
    int PAUSED = 10050;
    int RESULT_NOT_FOUND = 10060;
    int STOPPED = 10070;
    int TIME_OUT = 10080;
}